client/rest/src/main/java/org/elasticsearch/client/RestClient.java

```diff
@@ -563,6 +564,19 @@ private static boolean isSuccessfulResponse(int statusCode) {
         return statusCode < 300;
     }
 
+    /**
+     * Should an exception cause retrying the request?
+     */
+    private static boolean isRetryableException(Throwable e) {
+        if (e instanceof ExecutionException) {
+            e = e.getCause();
+        }
```

Comment:
I was going to say it'd be better to use the `try` in the caller, but seeing this thing, I get why you didn't.